armory/scenarios/poison.py

```diff
@@ -192,7 +196,23 @@ def poison_dataset(self):
                 np.array([]),
             )
 
+        self.n_poisoned = int(len(self.poison_index))
+        self.n_clean = (
+            len(self.y_clean) - self.n_poisoned
+        )  # self.y_clean is the whole pre-poison train set
+        poisoned = np.zeros_like(self.y_clean, np.bool)
+        poisoned[self.poison_index.astype(np.int64)] = True
+        self.probe.update(poisoned=poisoned)
+        self.hub.record("N_poisoned_train_samples", self.n_poisoned)
+        self.hub.record("N_clean_train_samples", self.n_clean)
+        self.train_set_class_labels = sorted(np.unique(self.y_clean))
```

Comment:
I think 208-212 belong better in load_train_dataset()

Reply:
The reason they are here is that the metrics are not set up until after `load_train_dataset` is called, do the `hub.record` call would fall on the floor.

Reply:
Ok makes sense.  Then all of the new additions to this function (`poison_dataset`) need to be copied to or otherwise inherited by poisoning_witches_brew.py, which overrides that function.  Currently, running witches_brew raises an `attribute error: no attribute self.train_set_class_labels`

Reply:
199-207 should be copied to witches_brew also, sorry if that wasn't clear:
```
        self.n_poisoned = int(len(self.poison_index))
        self.n_clean = (
            len(self.y_clean) - self.n_poisoned
        )  # self.y_clean is the whole pre-poison train set
        poisoned = np.zeros_like(self.y_clean, np.bool)
        poisoned[self.poison_index.astype(np.int64)] = True
        self.probe.update(poisoned=poisoned)
        self.hub.record("N_poisoned_train_samples", self.n_poisoned)
        self.hub.record("N_clean_train_samples", self.n_clean)
```